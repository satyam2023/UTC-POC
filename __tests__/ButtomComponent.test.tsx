import React from 'react';
import renderer from 'react-test-renderer';
import CustomButton from '../src/components/Button';
import { Image, Pressable, ViewStyle } from 'react-native';


describe('CustomButton Component', () => {
    test('component rendered', () => {
      const component = renderer.create(
        <CustomButton text="Click Me" />
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  
    test('isDisabled Props', () => {
      const component = renderer.create(
        <CustomButton text="Click Me" isDisabled={true} />
      );
      const button = component.root.findByType(Pressable);
      expect(button.props.disabled).toBe(true);
      expect(component.toJSON()).toMatchSnapshot();
    });
  
    test('button style', () => {
      const customStyle ={backgroundColor:'red'}
      const contStyle={height:10}
      const component = renderer.create(
        <CustomButton text="Click Me" buttonStyle={customStyle} />
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  
  
    test('image', () => {
      const imageSource = { uri: 'https://example.com/icon.png' };
      const component = renderer.create(
        <CustomButton text="Click Me" image={imageSource} />
      );
      const image = component.root.findByType(Image);
      expect(image.props.source).toEqual(imageSource);
    expect(component.toJSON()).toMatchSnapshot();
    });
  
    test('calls onPress ', () => {
      const onPressMock = jest.fn();
      const component = renderer.create(
        <CustomButton text="Click Me" onPress={onPressMock} />
      );
      const button = component.root.findByType(Pressable);
      button.props.onPress();
      expect(onPressMock).toHaveBeenCalled();
    });
  
    test('Image Style', () => {
      const customImageStyle = { width: 30, height: 30 };
      const imageSource = { uri: 'https://example.com/icon.png' };
      const component = renderer.create(
        <CustomButton text="Click Me" image={imageSource} imageStyle={customImageStyle} />
      );
      const image = component.root.findByType(Image);
      expect(image.props.style).toContainEqual(customImageStyle);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
