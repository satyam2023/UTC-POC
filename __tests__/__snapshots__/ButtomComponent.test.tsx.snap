// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton Component Image Style 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "blue",
        "borderRadius": 33,
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "center",
        "marginTop": "3%",
        "paddingHorizontal": 16,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      {
        "uri": "https://example.com/icon.png",
      }
    }
    style={
      [
        {
          "height": 20,
          "left": 16,
          "position": "absolute",
          "resizeMode": "contain",
          "width": 20,
        },
        {
          "height": 30,
          "width": 30,
        },
      ]
    }
  />
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;

exports[`CustomButton Component applies custom image style 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "blue",
        "borderRadius": 33,
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "center",
        "marginTop": "3%",
        "paddingHorizontal": 16,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      {
        "uri": "https://example.com/icon.png",
      }
    }
    style={
      [
        {
          "height": 20,
          "left": 16,
          "position": "absolute",
          "resizeMode": "contain",
          "width": 20,
        },
        {
          "height": 30,
          "width": 30,
        },
      ]
    }
  />
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;

exports[`CustomButton Component button style 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "blue",
        "borderRadius": 33,
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "center",
        "marginTop": "3%",
        "paddingHorizontal": 16,
        "width": "100%",
      },
      {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;

exports[`CustomButton Component component rendered 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "blue",
        "borderRadius": 33,
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "center",
        "marginTop": "3%",
        "paddingHorizontal": 16,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;

exports[`CustomButton Component image 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "blue",
        "borderRadius": 33,
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "center",
        "marginTop": "3%",
        "paddingHorizontal": 16,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      {
        "uri": "https://example.com/icon.png",
      }
    }
    style={
      [
        {
          "height": 20,
          "left": 16,
          "position": "absolute",
          "resizeMode": "contain",
          "width": 20,
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;

exports[`CustomButton Component isDisabled Props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "blue",
        "borderRadius": 33,
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "center",
        "marginTop": "3%",
        "paddingHorizontal": 16,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;

exports[`CustomButton Component renders correctly with a custom button style 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "blue",
        "borderRadius": 33,
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "center",
        "marginTop": "3%",
        "paddingHorizontal": 16,
        "width": "100%",
      },
      {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;

exports[`CustomButton Component renders correctly with an image 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "blue",
        "borderRadius": 33,
        "flexDirection": "row",
        "height": 56,
        "justifyContent": "center",
        "marginTop": "3%",
        "paddingHorizontal": 16,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      {
        "uri": "https://example.com/icon.png",
      }
    }
    style={
      [
        {
          "height": 20,
          "left": 16,
          "position": "absolute",
          "resizeMode": "contain",
          "width": 20,
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Click Me
  </Text>
</View>
`;
